DirectProgramming/DPC++FPGA/Tutorials/DesignPatterns/onchip_memory_cache/src/onchip_memory_cache.cpp

```diff
@@ -157,70 +114,70 @@ int main() {
         gold[b] = 0;
       }
       for (int i = 0; i < kInitNumInputs; ++i) {
-        int b = input_host[i] % kNumOutputs;
-        gold[b]++;
+        int hist_group = input_host[i] % kNumOutputs;
+        gold[hist_group]++;
       }
     }
 
     // Host accessor is now out-of-scope and is destructed. This is required
     // in order to unblock the kernel's subsequent accessor to the same buffer.
 
-    for (int i = 0; i < kNumRuns; i++) {
-      switch (i) {
-        case 0: {
-          std::cout << "Beginning run without on-chip memory caching.\n\n";
-          Histogram<false>(q, input_buf, output_buf, e);
-          break;
-        }
-        case 1: {
-          std::cout << "Beginning run with on-chip memory caching.\n\n";
-          Histogram<true>(q, input_buf, output_buf, e);
-          break;
-        }
-        default: {
-          Histogram<false>(q, input_buf, output_buf, e);
+    // iterate over the cache depths
+    for (int i = kMinCacheDepth; i < kMaxCacheDepth + 1; i++) {
+
+      std::cout << "Beginning run with cache depth " << i;
+      if (i == 0) { std::cout << " (no cache)"; }
+      std::cout << std::endl;
+
+      fpga_tools::UnrolledLoop<kMinCacheDepth, kMaxCacheDepth+1>([&](auto j) {
```

Comment:
Interesting workaround.
This is not an obvious technique that we expect the user to understand (and the underlying reasons) just by having a look at the code of this tutorial. 
I think you should add a comment explaining why you did that, and that it has nothing to do with how you should use the histogram

Reply:
Good point, I added a comment.  Have a look and please resolve this comment if you think it's sufficient.